source/uwp/Renderer/lib/CustomElementWrapper.cpp

```diff
@@ -0,0 +1,76 @@
+#include "pch.h"
+#include "CustomElementWrapper.h"
+
+using namespace Microsoft::WRL;
+using namespace ABI::AdaptiveCards::Uwp;
+
+namespace AdaptiveCards { namespace Uwp
+{
+
+bool CustomElementWrapper::GetSeparator() const
+{
+    ComPtr<IAdaptiveCardElement> cardElement;
+    m_cardElement.As(&cardElement);
+
+    boolean hasSeparator;
+    THROW_IF_FAILED(cardElement->get_Separator(&hasSeparator));
+    return hasSeparator;
+}
+
+void CustomElementWrapper::SetSeparator(const bool value)
+{
+    ComPtr<IAdaptiveCardElement> cardElement;
+    m_cardElement.As(&cardElement);
+
+    THROW_IF_FAILED(cardElement->put_Separator(value));
+}
+
+Spacing CustomElementWrapper::GetSpacing() const
+{
+    ComPtr<IAdaptiveCardElement> cardElement;
+    m_cardElement.As(&cardElement);
+
+    ABI::AdaptiveCards::Uwp::Spacing spacing;
+    THROW_IF_FAILED(cardElement->get_Spacing(&spacing));
+
+    return static_cast<Spacing> (spacing);
+}
+
+void CustomElementWrapper::SetSpacing(const Spacing value)
+{
+    ComPtr<IAdaptiveCardElement> cardElement;
+    m_cardElement.As(&cardElement);
+
+    THROW_IF_FAILED(cardElement->put_Spacing(static_cast<ABI::AdaptiveCards::Uwp::Spacing>(value)));
+}
+
+std::string CustomElementWrapper::GetId() const
+{
+    ComPtr<IAdaptiveCardElement> cardElement;
+    m_cardElement.As(&cardElement);
+
+    Wrappers::HString id; 
```

Comment:
>Wrappers::HString id;  [](start = 4, length = 22)

Is there a reason to use this over an HSTRING?
I ask because most of our other code uses HSTRINGs directly.

Reply:
HString will free the memory.  I will log an issue to follow up on other places we should be using it.

---
In reply to: [146940126](https://github.com/Microsoft/AdaptiveCards/pull/810#discussion_r146940126) [](ancestors = 146940126)

Reply:
#853

---
In reply to: [146953069](https://github.com/Microsoft/AdaptiveCards/pull/810#discussion_r146953069) [](ancestors = 146953069,146940126)